packages/nuxt/src/core/plugins/import-protection.ts

```diff
@@ -36,7 +36,7 @@ export const createImportProtectionPatterns = (nuxt: { options: NuxtOptions }, o
     ])
   }
 
-  for (const i of [/(^|node_modules\/)@nuxt\/(kit|test-utils)/, /(^|node_modules\/)nuxi/, /(^|node_modules\/)nitro(?:pack)?(?:-nightly)?(?:$|\/)(?!(?:dist\/)?runtime|types)/, /(^|node_modules\/)nuxt\/(config|kit|schema)/]) {
+  for (const i of [/(^|node_modules\/)@nuxt\/(kit|test-utils)/, /(^|node_modules\/)nuxi/, /(^|node_modules\/)nitro(?:pack)?(?:-nightly)?(?:$|\/)(?!(?:dist\/)?(?:presets|runtime|types))/, /(^|node_modules\/)nuxt\/(config|kit|schema)/]) {
```

Comment:
It would be better if we didn't depend on internal structure (even at cost of less import protection)

Reply:
perhaps nitro v3 would be a good time to revisit?

Reply:
While there is low chance for v2 to have major changes, internals changes can always happen. I believe we should instead protect only known subpath patterns (`nitropack` / `nitropack/kit` are main ones) + this logic should go into nitro itself for import protection (via unjs package)

Reply:
let's do it!